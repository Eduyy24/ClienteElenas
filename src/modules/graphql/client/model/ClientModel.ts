/**
 * Define el modelo para un Ciudad
 * @class
 */
export class CityModel {
  id = '';
  name = '';
}

/**
 * Define el modelo para un Estado
 * @class
 */
export class StateModel {
  id = '';
  name = '';
  shortCode = '';
  cities = Array<CityModel>(0);
}

/**
 * Define el modelo para address
 * @class
 */
 export class AddressModel {
  streetAddress = ''
  city = ''
  cityId = 0
  stateShortCode = ''
  stateId = 0
  country = ''
}


/**
 * Define el modelo de salida para un cliente
 * @class
 */
export class ClientOutputModel {
  id = '';
  firstName = '';
  lastName = '';
  cedula = '';
  address = '';
  cellphone = '';
  email = '';
  city = '';
  state?: StateModel;
}


/**
 * Define el modelo de entrada para un cliente
 * @class
 */
 export  class ClientInputModel {
  firstName = '';
  lastName = '';
  cedula = '';
  address = new AddressModel();
  cellphone = '';
  email = '';
}