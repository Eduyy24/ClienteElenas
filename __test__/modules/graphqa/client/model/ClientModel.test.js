import {ClientInputModel, ClientOutputModel, CityModel} from '../../../../../src/modules/graphql/client/model/ClientModel'

describe('ClientModel', () => {
  test('Generar ClientInputModel', () => {
    const city = new CityModel()
    expect(city).toMatchSnapshot()
  })
})