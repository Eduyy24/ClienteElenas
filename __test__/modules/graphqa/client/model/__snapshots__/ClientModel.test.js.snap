// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientModel Generar ClientInputModel 1`] = `
CityModel {
  "id": 0,
  "name": "",
}
`;
